CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -175,6 +181,42 @@ const char *_CFProcessPath(void) {
     }
     return __CFProcessPath;
 #else // TARGET_OS_BSD
+    char *argv0 = NULL;
+
+    // Get argv[0].
+#if defined(__OpenBSD__)
+    int mib[2] = {CTL_VM, VM_PSSTRINGS};
+    struct _ps_strings _ps;
+    size_t len = sizeof(_ps);
+
+    if (sysctl(mib, 2, &_ps, &len, NULL, 0) != -1) {
+        struct ps_strings *ps = _ps.val;
+        char *res = NULL;
+        res = realpath(ps->ps_argvstr[0], NULL);
+        argv0 = res? res: ps->ps_argvstr[0];
+    }
+#endif
+
+    if (!__CFProcessIsRestricted() && argv0 && argv0[0] == '/') {
+        _CFSetProgramNameFromPath(argv0);
+        return __CFProcessPath;
+    }
+
+    // Search PATH.
+    if (argv0) {
+      char *paths = getenv("PATH");
+      char *p = NULL;
+      while ((p = strsep(&paths, ":")) != NULL) {
+         char pp[PATH_MAX];
+         snprintf(pp, PATH_MAX, "%s/%s", p, argv0);
+         if (!__CFProcessIsRestricted() && access(pp, X_OK) == 0) {
+             _CFSetProgramNameFromPath(argv0);
+             return __CFProcessPath;
+         }
+      }
+    }
+
```

Comment:
Would this have issues if the executable was specified using a relative path ,eg `../some_executable` ? It would fail the first check but not necessarily be in the `PATH`

Reply:
Yes, that would not work if the realpath call failed and we fell back to argv[0]. We can call realpath a second time to ensure we get an absolute path. Added missing free calls as well.